paasta_tools/generate_all_deployments

```diff
@@ -6,18 +6,14 @@
 
 function delete_old_deployments {
     # Delete any deployments.json file that is older than an hour
-    find /nail/etc/services -iname deployments.json  -mmin +60 -delete
+    for service in $(comm -2 -3 <(find /nail/etc/services/ -maxdepth 1 -type d -printf %f\\n|tail -n +2|sort) <(paasta list|sort)); do
```

Comment:
I think that's a pretty good shout. It was my first instinct too. I guess even if a `deployments.json` is left over then we won't do anything with the service unless it appears in paasta list anyway.